sisl/io/orca/output.py

```diff
@@ -0,0 +1,343 @@
+        self._na = None
+        self._no = None
+
+    def readline(self, *args, **kwargs):
+        line = super().readline(*args, **kwargs)
+        if self._completed is None and "ORCA TERMINATED NORMALLY" in line:
+            self._completed = True
+        elif self._na is None and "Number of atoms" in line:
+            v = line.split()
+            self._na = int(v[-1])
+        elif self._no is None and "Number of basis functions" in line:
+            v = line.split()
+            self._no = int(v[-1])
+        return line
+
+    readline.__doc__ = SileORCA.readline.__doc__
+
+    @sile_fh_open()
+    def completed(self):
+        """ True if the full file has been read and "ORCA TERMINATED NORMALLY" was found. """
+        if self._completed is None:
+            completed = self.step_to("ORCA TERMINATED NORMALLY")[0]
+        else:
+            completed = self._completed
+        if completed:
+            self._completed = True
+        return completed
+
+    @property
+    @sile_fh_open()
+    def na(self):
+        """ Number of atoms """
+        if self._na is None:
+            f = self.step_to("Number of atoms")
+            if f[0]:
+                self._na = int(f[1].split()[-1])
+            else:
+                return None
+        return self._na
+
+    @property
+    @sile_fh_open()
+    def no(self):
+        """ Number of orbitals (basis functions) """
+        if self._no is None:
+            f = self.step_to("Number of atoms")
+            if f[0]:
+                self._no = int(f[1].split()[-1])
+            else:
+                return None
+        return self._no
+
+    @sile_fh_open()
+    def read_electrons(self, all=False):
+        """ Read number of electrons (alpha, beta)
+
+        Parameters
+        ----------
+        all : bool, optional
+            return electron numbers from all steps (instead of last)
+        """
+
+        def readE(itt, reread=True):
+            f = self.step_to("N(Alpha)", reread=reread)
+            if f[0]:
+                alpha = float(f[1].split()[-2])
+                beta = float(next(itt).split()[-2])
+            else:
+                return None
+            return alpha, beta
+
+        itt = iter(self)
+        E = []
+        e = readE(itt)
+        while e is not None:
+            E.append(e)
+            e = readE(itt, reread=False)
+
+        if all:
+            return np.array(E)
+        if len(E) > 0:
+            return np.array(E[-1])
+        return None
+
+    @sile_fh_open()
+    def read_charge(self, name='mulliken', projection='orbital', orbital=None,
+                    reduced=True, spin=False, all=False):
+        """ Reads from charge (or spin) population analysis
+
+        Parameters
+        ----------
+        name : {'mulliken', 'loewdin'}
+            name of the charge scheme to be read
+        projection : {'orbital', 'atom'}
+            whether to get orbital- or atom-resolved quantities
+        orbital : str, optional
```

Comment:
Shouldn't this be `orbitals` to follow other names.

Reply:
What about `orb_key` to make it distinct from the projection scheme?